examples/reshare/src/dkg/actor.rs

```diff
@@ -376,12 +350,132 @@ where
                             manager.process_block(epoch, block).await;
                         }
                     }
+
+                    // Wait to acknowledge until the block has been processed by the application.
+                    //
+                    // If we did not block on processing the block, marshal could continue processing finalized blocks and start
+                    // at a future block after restart (leaving the application in an unrecoverable state where we are beyond the last epoch height
+                    // and not willing to enter the next epoch).
+                    response.send(()).expect("response channel closed");
```

Comment:
The alternative to this approach would be to ask `marshal` for all blocks between what we've last processed and what we are being asked to process. I felt this was simpler (and what I use in Battleware).